src/app/bungie-api/http-client.ts

```diff
@@ -0,0 +1,219 @@
+import { PlatformErrorCodes, ServerResponse } from 'bungie-api-ts/common';
+import { HttpClientConfig, HttpClient } from 'bungie-api-ts/http';
+
+import { stringify } from 'querystring';
+import { delay } from 'app/utils/util';
+import _ from 'lodash';
+
+/**
+ * an error indicating a non-200 response code
+ */
+export class HttpStatusError extends Error {
```

Comment:
I remember that at some point this didn't work, but maybe modern TypeScript can handle it, or the fact that we're not transpiling classes makes it work (I assume we're not transpiling classes anymore?)

Reply:
not 100% sure what part wouldn't work but i would love help making sure something's not overlooked. it worked ok in testing when i spoofed http and api error codes

Reply:
If we're using classes directly without transforming them we should be OK. The Error constructor used to return a different instance that broke calls to "super"